README.md

```diff
@@ -147,23 +144,26 @@ functions** as
 [WebAssembly imports](https://webassembly.github.io/spec/core/syntax/modules.html#imports)
 (all of them defined in the `oak` module):
 
--   `channel_read: (i64, i32, i32) -> i32`: Reads data from the specified
-    channel.
+-   `channel_read: (i64, i32, i32) -> i32`: Reads a single message from the
+    specified channel. If the destination buffer is not large enough for the
+    entire message, then no data will be read and the required size will be
+    returned as a negative return value.
 
     *   arg 0: Handle to channel receive half
     *   arg 1: Destination buffer address
     *   arg 2: Destination buffer size in bytes
-    *   return 0: Number of bytes read
+    *   return 0: Number of bytes read (if positive), otherwise minus
+        the number of bytes required to read the message.
 
     Similar to
     [`zx_channel_read`](https://fuchsia.googlesource.com/fuchsia/+/refs/heads/master/zircon/docs/syscalls/channel_read.md)
     in Fuchsia.
 
--   `channel_write: (i64, i32, i32) -> i32`: Writes data to the specified
-    channel.
+-   `channel_write: (i64, i32, i32) -> i32`: Writes a single message to the
```

Comment:
Arguably the number of bytes written is completely irrelevant, since writes either succeed or fail for the entire message? Another argument for having syscalls return status code I guess :)